charms/istio-pilot/src/gateway.yaml.j2

```diff
@@ -11,7 +12,27 @@ spec:
   servers:
   - hosts:
     - '*'
+{% if not secret_name %}
     port:
       name: http
       number: 80
       protocol: HTTP
+{% else %}
+    port:
+      name: https
+      number: 443
+      protocol: HTTPS
+    tls:
+      mode: SIMPLE
+      credentialName: {{ secret_name }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ secret_name }}
+  namespace: {{ model_name }}
+type: kubernetes.io/tls
+data:
+  tls.crt: {{ ssl_crt }}
+  tls.key: {{ ssl_key }}
+{% endif %}
```

Comment:
This seems a bit unexpected that `gateway.yaml.j2` has the side effect of sometimes including a secret.  Can we put the secret in a separate `gateway-secret.yaml.j2` or similar, and selectively apply that file?

Reply:
Agreed, I will test it now